cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1007,6 +1007,50 @@ def to_numpy_by_lag(self) -> Tuple[Dict[int, numpy.ndarray], List[str]]:
         assert self.variables is not None
         return adjacency_matrices, self.variables
 
+    def get_nodes_at_lag(self, time_lag: int = 0) -> List[TimeSeriesNode]:
+        """Return all nodes at time delta `time_lag`."""
+        return [node for node in self.get_nodes() if node.time_lag == time_lag]
+
+    def get_contemporaneous_nodes(self, node: NodeLike) -> List[TimeSeriesNode]:
+        """Return all nodes that are contemporaneous to the provided node."""
```

Comment:
`contamperaneous (i.e. have the same time_lag)`

Reply:
could this use get_nodes_at_lag?

`get_nodes_at_lag(node.time_lag).filter(lambda n: n.identifier != node.identifier)`